pkg/apis/core/validation/validation_test.go

```diff
@@ -25860,9 +25874,47 @@ func TestValidatePodResize(t *testing.T) {
 			new: mkPod(core.ResourceList{}, getResources("200m", "0", "1Gi", "")),
 			err: "",
 		},
+		{
+			test:         "cpu limit change for sidecar containers",
+			old:          mkPodWithInitCtrs(core.ResourceList{}, getResources("100m", "0", "1Gi", "")),
+			new:          mkPodWithInitCtrs(core.ResourceList{}, getResources("200m", "0", "1Gi", "")),
+			isSideCarCtr: true,
+			err:          "",
+		}, {
+			test:         "memory limit change for sidecar containers",
+			old:          mkPodWithInitCtrs(core.ResourceList{}, getResources("100m", "200Mi", "", "")),
+			new:          mkPodWithInitCtrs(core.ResourceList{}, getResources("100m", "100Mi", "", "")),
+			isSideCarCtr: true,
+			err:          "",
+		}, {
+			test:         "storage limit change for sidecar containers",
+			old:          mkPodWithInitCtrs(core.ResourceList{}, getResources("100m", "100Mi", "2Gi", "")),
+			new:          mkPodWithInitCtrs(core.ResourceList{}, getResources("100m", "100Mi", "1Gi", "")),
+			isSideCarCtr: true,
+			err:          "spec: Forbidden: cpu and memory resources for only sidecar containers are mutable",
+		}, {
+			test:         "cpu request change for sidecar containers",
+			old:          mkPodWithInitCtrs(getResources("200m", "0", "", ""), core.ResourceList{}),
+			new:          mkPodWithInitCtrs(getResources("100m", "0", "", ""), core.ResourceList{}),
+			isSideCarCtr: true,
+			err:          "",
+		}, {
+			test:         "memory request change for sidecar containers",
+			old:          mkPodWithInitCtrs(getResources("0", "100Mi", "", ""), core.ResourceList{}),
+			new:          mkPodWithInitCtrs(getResources("0", "200Mi", "", ""), core.ResourceList{}),
+			isSideCarCtr: true,
+			err:          "",
+		}, {
+			test:         "storage request change for sidecar containers",
+			old:          mkPodWithInitCtrs(getResources("100m", "0", "1Gi", ""), core.ResourceList{}),
+			new:          mkPodWithInitCtrs(getResources("100m", "0", "2Gi", ""), core.ResourceList{}),
+			isSideCarCtr: true,
+			err:          "spec: Forbidden: cpu and memory resources for only sidecar containers are mutable",
+		},
 	}
 
 	for _, test := range tests {
+		featuregatetesting.SetFeatureGateDuringTest(t, utilfeature.DefaultFeatureGate, features.SidecarContainers, test.isSideCarCtr)
```

Comment:
This feature is already enabled at the start of `TestValidatePodResize` func.

Reply:
Thanks. Made the change.